tests/virt_autotest/esxi_open_vm_tools.pm

```diff
@@ -0,0 +1,218 @@
+            sleep(10);
+            $num--;
+        }
+    }
+    # Guest Shutdown
+    if (esxi_vm_power_getstate($vm_id) =~ m/Powered on/) {
+        record_info('Guest Shutdown');
+        esxi_vm_power_shutdown($vm_id);
+        my $num = 6;
+        while ($num && console('svirt')->run_cmd("! ping -c1 $vm_ip")) {
+            sleep(10);
+            $num--;
+        }
+    }
+}
+
+sub networking_tests {
+    my ($vm_id, $vm_ip) = @_;
+
+    if (esxi_vm_power_getstate($vm_id) =~ m/Powered off/) {
+        esxi_vm_power_on($vm_id);
+        my $num = 6;
+        while ($num && console('svirt')->run_cmd("ping -c1 $vm_ip")) {
+            sleep(10);
+            $num--;
+        }
+    }
+
+    my $vswitch_name = esxi_vm_network_binding($vm_id);
+    record_info('VSwitch Name', $vswitch_name);
+
+    reset_consoles;
+    select_console("sut");
+    assert_screen("grub2", 90);
+    send_key 'ret';
+    select_console("root-console", timeout => 180);
+
+    assert_script_run("ping -c3 openqa.suse.de");
+    assert_script_run("ping -c3 www.suse.com");
+}
+
+sub clock_sync_tests {
+    my ($vm_name, $vm_id, $vm_ip) = @_;
+    my ($g_datetime, $h_datetime, $g_difftime);
+
+    # Enable timesync service in guest VM
+    if (script_output("vmware-toolbox-cmd timesync status", proceed_on_failure => 1, timeout => 90) == "Disabled") {
+        $g_datetime = script_output("date -u +'\%Y-\%m-\%d \%H:\%M:\%S'");
+        record_info('Guest current time before syncing up', $g_datetime);
+        assert_script_run("vmware-toolbox-cmd timesync enable", timeout => 90);
+        $h_datetime = get_host_timestamp();
+        record_info('Host time', $h_datetime);
+        $g_datetime = script_output("date -u +'\%Y-\%m-\%d \%H:\%M:\%S'");
+        record_info('Guest time after syncing up with host', $g_datetime);
+    }
+
+    my $h_timesec = get_timelc($h_datetime);
+    my $g_timesec = get_timelc($g_datetime);
+    my $diff_sec  = int($g_timesec - $h_timesec);
+
+    record_info('Diff secs', $diff_sec);
+    if ($diff_sec <= 10) {
+        record_info("Clock synchronization is successful.");
+    } else {
+        die "Clock synchronization failed.";
+    }
+
+    select_console('svirt');
+
+    # Disable clock sync tests
+    if (esxi_vm_power_getstate($vm_id) =~ m/Powered on/) {
+        esxi_vm_power_shutdown($vm_id);
+        my $num = 6;
+        while ($num && console('svirt')->run_cmd("! ping -c1 $vm_ip")) {
+            sleep(10);
+            $num--;
+        }
+    }
+
+    disable_all_clock_synchronization($vm_name);
+
+    if (esxi_vm_power_getstate($vm_id) =~ m/Power off/) {
+        esxi_vm_power_on($vm_id);
+        my $num = 6;
+        while ($num && console('svirt')->run_cmd("ping -c1 $vm_ip")) {
+            sleep(10);
+            $num--;
+        }
+    }
+
+    reset_consoles;
+    select_console('sut');
+    assert_screen("grub2", 90);
+    send_key 'ret';
+    select_console("root-console", timeout => 180);
```

Comment:
This code block(L190~194) seems to repeat, suggest to put it in a function.

Reply:
OK, I've put it in a function called "login_vm_console".